### DISCUSSION TITLE
json api key/value support for variables

### DISCUSSION THREAD

**Discussion by stefan-g** [grafana/grafana-json-datasource#88] [Q&A] [answered]
I would like to use the json api to defines some variables in grafana from a json data source. The variable in grafana should be defined as key  values pair but as i've seen in the documentation it does only support one field. Did i missed something or is it not possible to get a key and a value from a json? If it's not possible i think it would be a nice feature.

**Comment by marcusolsson** [comment, marked as answer]
I'm not sure I understand the question. Dashboard variables in can only be strings or numbers, which is a limitation in Grafana rather than this plugin. 

Could you give an example of how you'd want to use key/values in variables?

**Reply by stefan-g** [reply]
Yes, I have a metric label_value that is defined as a number (like 100,101,102,....). Each number has an assigned text. In grafana i would like to define a more userfriendly variable that shows the text instead of the number. Key/value pairs can be defined as customer variables (format is myKey:myValue, ....). I might have found a solution with regex but not sure if it works yet...

**Reply by marcusolsson** [reply]
If you'd like to be able to set an optional label for the variable value, there's an [open feature request](https://github.com/marcusolsson/grafana-json-datasource/issues/79) for this. Please add your thoughts in that thread if you think it would be helpful for your use case 🙏

**Reply by stefan-g** [reply]
Thanks for your answer. I'm looking for the same feature.
